tools/jvm-options-parser/src/main/java/org/logstash/launchers/JvmOptionsParser.java

```diff
@@ -404,6 +405,23 @@ private static Optional<String> jvmOptionFromLine(final int javaMajorVersion, fi
         return Optional.empty();
     }
 
+    private static final Pattern ENV_VAR_PATTERN = Pattern.compile("\\$\\{([a-zA-Z_.][a-zA-Z0-9_.]*)(?::([^}]*))?\\}");
```

Comment:
```suggestion
    private static final Pattern ENV_VAR_PATTERN = Pattern.compile("\\$\\{([a-zA-Z_][a-zA-Z0-9_]*)(?::([^}]*))?\\}");
```
Lets remove `.` as they are not in env var names? https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html

Reply:
We support`.` in env var, which is the same as Elasticsearch and the current Logstash.
For example, `docker run -d --name=es -p 9200:9200 -e "discovery.type=single-node" -e xpack.security.enabled=false -e xpack.security.http.ssl.enabled=false docker.elastic.co/elasticsearch/elasticsearch:8.17.0`

The moost important is we want to be consistent to the existing Logstash implementation. So, we should keep `.`

Reply:
TIL. I didnt think that it was possible to set environment variables with dots in the names. I verified that if you can get it set (i tricked a bash shell in to doing it with `env`) java can in fact read it. 

```
➜  /tmp cat EnvVarDotTest.java
import java.util.Map;

public class EnvVarDotTest {
    public static void main(String[] args) {
        Map<String, String> env = System.getenv();
        env.forEach((key, value) -> {
            if (key.contains(".")) {
                System.out.println(key + "=" + value);
            }
        });
    }
}
➜  /tmp env test.var=hello java EnvVarDotTest
test.var=hello
```
Might be worth adding a test with a name that contains a dot. If nothing else but to alert that we do intend to support this (IMO kind of surprising) behavior.